src/types/api.types.ts

```diff
@@ -25,6 +25,7 @@ export type ChatActions =
 
 export type SendOptions =
   | ISendOptions
+  | ISendText
```

Comment:
**Suggestion:** It's not clear to me that this shouldn't go in another `I.*Options` object instead, as this will make it so the only options you can specify would be `link_preview_options`.

I think this may need to go on `SendOptions` as a whole, documented as "only working with `the SendMessage endpoint`:

```ts
export type SendOptions = (
  | ISendOptions
  | ISendText
  | ISendAudioOptions
  | ISendAnimationOptions
  | ISendContactOptions
  | ISendDiceOptions
  | ISendDocumentOptions
  | ISendLocationOptions
  | ISendMediaGroupOptions
  | ISendPhotoOptions
  | ISendPollOptions
  | ISendVenueOptions
  | ISendVideoOptions
  | ISendVideoNoteOptions
  | ISendVoiceOptions) & Readonly<{
    /** Only works for send/edit messages */
    link_preview_options?: LinkPreviewOptions
  }>
```

Then, in `Api.ts`, we need to make sure that we delete these options when not calling `sendMessage` (or probably `editMessage` if it's called there, too)